config/config.go

```diff
@@ -290,14 +291,35 @@ func determineTreeRoot(v *viper.Viper, cfg *Config, logger *log.Logger) error {
 		return errors.New("at most one of tree-root, tree-root-cmd or tree-root-file can be specified")
 	}
 
-	// set git-based tree root command if the walker is git and no tree root has been specified
-	if cfg.Walk == walk.Git.String() && count == 0 {
-		cfg.TreeRootCmd = "git rev-parse --show-toplevel"
+	// only attempt the following if no tree root has been provided
+	if count == 0 {
+		setGitRootCmd := func() {
+			cfg.TreeRootCmd = git.TreeRootCmd
 
-		logger.Infof(
-			"git walker enabled and tree root has not been specified: defaulting tree-root-cmd to '%s'",
-			cfg.TreeRootCmd,
-		)
+			logger.Infof(
+				"git repo detected and tree root has not been specified: defaulting tree-root-cmd to '%s'",
+				cfg.TreeRootCmd,
+			)
+		}
+
+		switch {
+		case cfg.Walk == walk.Auto.String():
+			// check if the directory containing the config file is a git repo
+			isGitRepo, err := git.IsInsideWorktree(cfg.WorkingDirectory)
+			if err != nil {
+				return fmt.Errorf("failed to check if working directory is a git repo: %w", err)
+			}
+
+			// if it is, set the default git tree root command
+			if isGitRepo {
+				setGitRootCmd()
+			}
```

Comment:
Would it be simpler, instead of a) checking if we're in a git directory and b) setting a default `tree-root-cmd` if we are, to instead do something like:

If no `tree-root`, `tree-root-file`, or `tree-root-cmd` are set; then a) attempt to set `tree-root` directly by running `git rev-parse --show-toplevel` b) silently fail if this fails.

Without explicitly checking that we're actually in a git repo via `git rev-parse --is-inside-work-tree`.

It seems to me, the logic of checking "are we in a git repo" is the same logic we would need to attempt to run `git rev-parse --show-toplevel` without knowing (or caring) if we're actually in a git repo.

Reply:
It's a fair simplification. I'll have a look in a bit.

Reply:
Much simpler now. I'll look at beefing up test coverage this evening.